plugins/modules/identity/keycloak/keycloak_authentication.py

```diff
@@ -0,0 +1,382 @@
+                        if new_exec[key] is None:
+                            exclude_key.append(key)
+                    # Compare the executions to see if it need changes
+                    if not is_struct_included(new_exec, existing_executions[exec_index], exclude_key) or exec_index != new_exec_index:
+                        changed = True
+                elif "providerId" in new_exec:
+                    kc.create_execution(new_exec, flowAlias=flow_alias_parent, realm=realm)
+                    changed = True
+                elif "displayName" in new_exec:
+                    kc.create_subflow(new_exec["displayName"], flow_alias_parent, realm=realm)
+                    changed = True
+                if changed:
+                    # Get existing executions on the Keycloak server for this alias
+                    existing_executions = kc.get_executions_representation(config, realm=realm)
+                    exec_index = find_exec_in_executions(new_exec, existing_executions)
+                    if exec_index != -1:
+                        # Update the existing execution
+                        updated_exec = {
+                            "id": existing_executions[exec_index]["id"]
+                        }
+                        for key in new_exec:
+                            # create the execution configuration
+                            if key == "authenticationConfig":
+                                kc.add_authenticationConfig_to_execution(updated_exec["id"], new_exec["authenticationConfig"], realm=realm)
+                            # remove potential flowAlias for the next API call
+                            elif key != "flowAlias":
+                                updated_exec[key] = new_exec[key]
+                        if "requirement" in new_exec:
+                            kc.update_authentication_executions(flow_alias_parent, updated_exec, realm=realm)
+                        diff = exec_index - new_exec_index
+                        kc.change_execution_priority(updated_exec["id"], diff, realm=realm)
+        return changed
+    except Exception as e:
+        kc.module.fail_json(msg='Could not create or update executions for authentication flow %s in realm %s: %s'
+                            % (config["alias"], realm, str(e)))
+
+
+def main():
+    """
+    Module execution
+    :return:
+    """
+    argument_spec = keycloak_argument_spec()
+    meta_args = dict(
+        realm=dict(type='str', required=True),
+        alias=dict(type='str', required=True),
+        providerId=dict(type='str'),
+        description=dict(type='str'),
+        copyFrom=dict(type='str'),
+        authenticationExecutions=dict(type='list', elements='dict',
+                                      options=dict(
+                                          providerId=dict(type='str'),
+                                          displayName=dict(type='str', required=False),
```

Comment:
Another nit-picking: `required=False` is redundant.

Reply:
resolved by 'ca6b4fe'